lighthouse-core/lib/traces/tracing-processor.js

```diff
@@ -200,22 +190,44 @@ class TraceProcessor {
   }
 
   /**
-   * Provides the top level events on the main thread.
+   * Provides the top level events on the main thread with timestamps in ms relative to navigation
+   * start.
    * @param {!TraceOfTabArtifact} tabTrace
    * @param {number=} startTime Optional start time (in ms) of range of interest. Defaults to trace start.
    * @param {number=} endTime Optional end time (in ms) of range of interest. Defaults to trace end.
    * @return {!Array<{start: number, end: number, duration: number}>}
    */
   static getMainThreadTopLevelEvents(tabTrace, startTime = -Infinity, endTime = Infinity) {
-    // Find the main thread via the first TracingStartedInPage event in the trace
-    const startEvent = tabTrace.startedInPageEvt;
-    const mainThreadEvents = tabTrace.processEvents.filter(e => e.tid === startEvent.tid);
-
-    return mainThreadEvents.filter(event => {
-      return event.name === SCHEDULABLE_TASK_TITLE &&
-          (event.ts + event.dur) > startTime &&
-          event.ts < endTime;
-    });
+    let lastEvt = {start: -Infinity, end: -Infinity};
+    return tabTrace.processEvents.reduce((events, event) => {
+      if (event.name !== SCHEDULABLE_TASK_TITLE || !event.dur) return events;
+
+      const start = (event.ts - tabTrace.navigationStartEvt.ts) / 1000;
+      const end = (event.ts + event.dur - tabTrace.navigationStartEvt.ts) / 1000;
+      if (start > endTime || end < startTime) return events;
+
+      const currentEvt = {
+        start,
+        end,
+        duration: event.dur / 1000,
+        events: [event],
+      };
+
+      if (currentEvt.end < lastEvt.end) {
+        // The current event is completely inside the previous event, add to previous
+        lastEvt.events.push(event);
+      } else if (currentEvt.start < lastEvt.end) {
```

Comment:
this is possible? 

we're in a single process, though admittedly not a single thread. since we're only looking at these scheduleable tasks, i dont think they should overlap..